File: programming-concepts/good-programming-practice/java/identifiers.java
/*
Raspberry Pi Foundation
Developed as part of Ada Computer Science

Usage licensed under CC BY-NC-SA 4.0

Note: This file is designed to be copied out and compiled on your machine.
In order for the program to compile properly in some IDEs, you need to ensure that the
class filename is the same as the class name with the main method in it.

To run this file you need to:
1. Copy the contents
2. Create a Java project in the IDE of your choice (Eclipse, for example)
3. Create a new Java Class that uses the same name as the class containing the main method in this program
4. Paste the contents into the new Java Class
5. Save and run the program
*/

class GoodProgrammingPractice {
    static final double GRAVITY_ON_EARTH = 9.81;
    static final double GRAVITY_ON_MARS = 3.711;

    // The main method is the entry point for all Java programs
    public static void main(String[] args) {
        // Execute both subroutines and output the results
        double weight = 35.7;
        System.out.println(conv(weight));
        System.out.println(calculateWeightOnMars(weight));
    }


    // First example
    public static double conv(double a) {
        double b = a / 9.81 * 3.711;
        return b;
    }


    // Second example
    // Convert a weight on Earth to the weight on Mars based on gravity
    public static double calculateWeightOnMars(double weightOnEarth) {
        double weightOnMars = weightOnEarth / GRAVITY_ON_EARTH * GRAVITY_ON_MARS;
        return weightOnMars;
    }


}

File: programming-concepts/good-programming-practice/c/identifiers.c
#include <stdio.h>


//first example

int conv(int a)
{
    return a / 9.81 * 3.17;
}

//second example
float GRAVITY_ON_EARTH = 9.81;
float GRAVITY_ON_MARS = 3.711;

float calculate_weight_on_mars (float weight_on_earth)
{
    return (weight_on_earth / GRAVITY_ON_EARTH) * GRAVITY_ON_MARS;
}

//driver code
int main(void)
{
    int weight_on_earth = 80;
    int weight_on_mars = conv(weight_on_earth);
    printf("Weight on Earth: %d\n", weight_on_earth);
    printf("Weight on Mars: %d\n", weight_on_mars);

    printf("\n");

    float weight_on_earth2 = 80;
    float weight_on_mars2 = calculate_weight_on_mars(weight_on_earth2);
    printf("Weight on Earth: %.2f\n", weight_on_earth2);
    printf("Weight on Mars: %.2f\n", weight_on_mars2);
}

// Credit: Michael Parker https://mastodon.social/@michaelparker